java/src/main/java/sh/platform/languages/SampleCodeSupplier.java

```diff
@@ -0,0 +1,52 @@
+package sh.platform.languages;
+
+import sh.platform.languages.compiler.JavaCompilerProvider;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class SampleCodeSupplier implements Supplier<Map<SamplesAvailable, SampleCode>> {
+
+    private static final Logger LOGGER = Logger.getLogger(SampleCodeSupplier.class.getName());
+
+    private final JavaCompilerProvider compiler = new JavaCompilerProvider();
+
+    private final Map<SamplesAvailable, SampleCode> cached = new EnumMap<>(SamplesAvailable.class);
+
+    {
```

Comment:
I am confused by this syntax... This code block doesn't seem to be part of any method?  How does it run?  (Likely a bit of Java-fu I lack.)

Reply:
It will start when the JVM starts this instance.

Reply:
Isn't that what a constructor is?  Or is it like a static constructor that runs only once?

Reply:
Think about a constructor in other languages, but, with higher priority on execution.
What I'm doing:
Compile the code if get any issue stop and don't even start the constructor.